### PR TITLE
Bug 1622339 -  Move android-sdk installation into a toolchain task

### PR DESCRIPTION
**Pull request by JohanLorenzo**
## Checklist

- [ ] Confirm the **acceptance criteria** is fully satisfied in the issue(s) this PR will close
- [ ] Add thorough **tests** or an explanation of why it does not
- [ ] Add a **CHANGELOG entry** if applicable
- [ ] Add **QA labels** on the associated issue (not this PR; `qa-ready` or `qa-notneeded`)

### CODE REVIEW

**Comment by JohanLorenzo** [comment]
The UI test issue doesn't seem related to these changes, to me.

File: taskcluster/firefox_tv_taskgraph/job.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+def _extract_gradlew_command(run):
+    pre_gradle_commands = run.pop("pre-gradlew", [])
+    pre_gradle_commands += [
+        _generate_dummy_secret_command(secret) for secret in run.pop("dummy-secrets", [])
+    ]
+    pre_gradle_commands += [
+        _generate_secret_command(secret) for secret in run.get("secrets", [])
+    ]
+
+    gradle_command = ["./gradlew"] + run.pop("gradlew")
```
Comment (tomprince) [inline taskcluster/firefox_tv_taskgraph/job.py:96]: Looking at this, I'm curious if there is any value in making this gradle-specific, and having the pre- and post- bits. (particularly as I see one of the post commands is also using gradle).

File: taskcluster/ci/build/kind.yml [hunk 2]
```diff
@@ -5,59 +5,62 @@
...
-        git checkout {tag}
-        yes | sdkmanager --licenses
-        python tools/taskcluster/get-sentry-token.py
-        python tools/taskcluster/get-pocket-token.py
-        ./gradlew --no-daemon clean test assembleSystemRelease
+    pr:
+        run:
+            gradlew: [-PisPullRequest, clean, assembleSystem, assembleAndroidTest, lint, checkstyle, ktlint, pmd, detekt, test]
+            post-gradlew:
+                - [./gradlew, -Pcoverage, jacocoSystemDebugTestReport]
```
Comment (tomprince) [inline taskcluster/ci/build/kind.yml:38]: I think these lists might be a little clearer to read with quotes.

File: taskcluster/firefox_tv_taskgraph/job.py (hunk 1 above)
Comment (JohanLorenzo) [reply taskcluster/firefox_tv_taskgraph/job.py:96]: Hmmm, that's a great question! Let's see.

The `gradlew` kind was first introduced in [August 2019](https://github.com/mozilla-mobile/reference-browser/pull/737/files#diff-81790a570b45107d44ea76233a7754e8R33). `post-gradlew` was added [the following month](https://github.com/mozilla-mobile/reference-browser/pull/892/files#diff-24c9b9fc54c62f21ea7365d5b268f925R54). `pre-gradlew` was then added [3 weeks later](https://github.com/mozilla-mobile/fenix/pull/5488/files#diff-d4885d206919bcc0e772ccdb9618c071R18).

`run-commands` first appeared in [November 2019](https://github.com/mozilla-mobile/fenix/pull/6517/files#diff-d4885d206919bcc0e772ccdb9618c071R42), in order to support non-gradle tasks.

We could convert `gradlew` flavors into `run-commands` since the former grew organically. I actually wondered if I could combine [that singular post-command](https://github.com/mozilla-mobile/firefox-tv/pull/2984/files#diff-ec565f108a3a09f10c613b7dea5ee557R38) into the big gradlew list. Although, I didn't want to break anything so I preferred to keep 2 separate commands. 

One thing I like about keeping `gradlew` is: https://github.com/mozilla-mobile/reference-browser/pull/1117/files#diff-81790a570b45107d44ea76233a7754e8R99-R105. By keeping this flavor around, we allow each `gradlew` call to be rewired to use specific maven repositories. This is something we cannot easily do with `run-commands`.

To sum up, I'd be in favor of keeping `gradlew` and try to merge the one post-command that still uses `gradlew` in a potential followup. For the context, firefox-tv is a deprecated project. I've just been backporting important taskgraph patches. I currently don't see this gradlew merge as important. How does that sound to you @tomprince?

**Comment by JohanLorenzo** [comment]
@mcomella hey! Just a heads-up that I'm going to need your merge power soon (just like I did in https://github.com/mozilla-mobile/firefox-tv/pull/2983#issuecomment-603111874) 🙂

**Comment by mcomella** [comment]
r+ from TC folks: lgtm. Merging...
